src/utils/form.ts

```diff
@@ -12,3 +14,148 @@ export const hiddenInputStyle = `
   -webkit-appearance: none !important;
   z-index: -1 !important;
 `;
+
+interface FormDataEvent extends Event {
+  formData: FormData;
+}
+
+/**
+ * Defines interface for components that are form-compatible.
+ *
+ * Along with the interface, use the matching form utils to help set up the component behavior.
+ */
+export interface FormAssociated<T = any> {
+  /**
+   * The host element.
+   */
+  readonly el: HTMLElement;
+
+  /**
+   * The form this component is associated with.
+   */
+  formEl: HTMLElement;
+
+  /**
+   * The name used to submit the value to the associated form.
+   *
+   * Note that this prop should use the @Prop decorator.
+   *
+   * @default ""
+   */
+  name: string;
+
+  /**
+   * This form component's value.
+   *
+   * Note that this prop should use the @Prop decorator.
+   */
+  value: T;
+
+  /**
+   * The initial value for this form component.
+   *
+   * When the form is reset, the value will be set to this property.
+   */
+  initialValue: T;
+
+  /**
+   * Hook for components to provide custom form reset behavior.
+   */
+  onFormReset?(): void;
```

Comment:
nice. I like the `on` name here. I'll update my interface names to be consistent with these.